src/Microsoft.Extensions.Logging.Abstractions/LoggerMetricsExtensions.cs

```diff
@@ -0,0 +1,16 @@
+namespace Microsoft.Extensions.Logging
+{
+    public static class LoggerMetricsExtensions
+    {
+        public static void RecordMetric(this ILogger logger, string name, double value) => logger.RecordMetric(new Metric(name, value));
+
+        public static void RecordMetric(this ILogger logger, Metric metric)
```

Comment:
Do we need some high performance alternative like `DefineMessage` is for usual logging? For example to prevent the need to look up metric by name inside loggers for aggregation.